contracts/src/CrossChainApplications/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -0,0 +1,263 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {IWarpMessenger} from "@subnet-evm-contracts/interfaces/IWarpMessenger.sol";
+import {INativeMinter} from "@subnet-evm-contracts/interfaces/INativeMinter.sol";
+import {INativeTokenDestination} from "./INativeTokenDestination.sol";
+import {ITokenSource} from "./ITokenSource.sol";
+import {ITeleporterMessenger, TeleporterFeeInfo, TeleporterMessageInput} from "../../Teleporter/ITeleporterMessenger.sol";
+import {ITeleporterReceiver} from "../../Teleporter/ITeleporterReceiver.sol";
+import {SafeERC20TransferFrom} from "../../Teleporter/SafeERC20TransferFrom.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+// We need IAllowList as an indirect dependency in order to compile.
+// solhint-disable-next-line no-unused-import
+import {IAllowList} from "@subnet-evm-contracts/interfaces/IAllowList.sol";
+
+contract NativeTokenDestination is
+    ITeleporterReceiver,
+    INativeTokenDestination,
+    ReentrancyGuard
+{
+    // The address where the burned transaction fees are credited.
+    address public constant BURNED_TX_FEES_ADDRESS = 0x0100000000000000000000000000000000000000;
+    // Designated Blackhole Address. Tokens are sent here to be "burned" before sending an unlock
+    // message to the source chain. Different from the burned tx fee address so they can be
+    // tracked separately.
+    address public constant BLACKHOLE_ADDRESS = 0x0100000000000000000000000000000000000001;
```

Comment:
I commented thsi previously, but can't find the previous comment. How are these addresses determined are they predefined for subnet-evm where tx fees are already burned to that address? If so, can you leave a comment.

Reply:
We should link to something like https://github.com/ava-labs/subnet-evm/blob/03c521954ec737c759e3fdd07c008d45cf7a292c/constants/constants.go